src/utils/UTCDate/index.ts

```diff
@@ -3,18 +3,52 @@
  */
 export default class UTCDate {
 
+  /** The  */
+  private readonly date: Date;
+
   /** Old date, will always be older than last modified */
   public static OLD_DATE = '1970-01-01 00:00:00Z';
 
+  public constructor(utcDateString: string) {
+    if (Number.isNaN(Date.parse(utcDateString))) throw new Error('Invalid date string');
+    this.date = new Date(utcDateString);
+  }
+
+  /**
+   * toString method
+   *
+   * @override
+   */
+  public toString() {
+    return UTCDate.convertToString(this.date);
+  }
+
   /** Convert a date string to Date (UTC) */
   public static convertToDate(dateString: string) {
     const utcDateString = `${dateString}Z`;
     return new Date(utcDateString);
   }
 
+  private static pad(digits: number, upperLimit = 10) {
+    return digits < upperLimit ? '0' + digits : digits;
+  }
+
   /** Get current date UTC as a string */
   public static getCurrentDate(): string {
-    return new Date().toUTCString();
+    return UTCDate.convertToString(new Date());
+  }
+
+  private static convertToString(date: Date) {
```

Comment:
could use moment?

Reply:
I think James did the work for this on his branch already